Checked that the test fails without the code changes:

```
./test/functional/rpc_signrawtransaction.py 
2020-12-04T14:55:24.933000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_q8rf0stk
2020-12-04T14:55:26.373000Z TestFramework (INFO): Try with a P2PKH script as the witnessScript
2020-12-04T14:55:26.389000Z TestFramework (INFO): Try with a P2PK script as the witnessScript
2020-12-04T14:55:26.741000Z TestFramework (INFO): Test signing a fully signed transaction does nothing
2020-12-04T14:55:26.946000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/marco/workspace/btc_bitcoin_core/test/functional/test_framework/test_framework.py", line 112, in main
    self.run_test()
  File "/home/marco/workspace/btc_bitcoin_core/./test/functional/rpc_signrawtransaction.py", line 227, in run_test
    self.test_fully_signed_tx()
  File "/home/marco/workspace/btc_bitcoin_core/./test/functional/rpc_signrawtransaction.py", line 158, in test_fully_signed_tx
    assert_equal(signedtx2["complete"], True)
  File "/home/marco/workspace/btc_bitcoin_core/test/functional/test_framework/util.py", line 46, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(False == True)
```

Approach ACK a4b56df42e42ad304571ed82648893484accfa5e 🥔

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Approach ACK a4b56df42e42ad304571ed82648893484accfa5e 🥔
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUho6AwAy4I+vrdOq7pk7ML0Dehz3qSStFH7iRuDJgJ8Y51p1BiD5KyVeT0gh0vj
ZJ3Cidqnw5BF6oDhFRGWCu1SZyy3yiNFBUqR/HhqSprpSenYcm2BXvCHuKl2BjNh
DqDMsER379Joc3vdIqFBDspyQceUS+4gvkqsgn9tct1kz2Fpzw0ZXo5FPKpj+7Pd
hs7lvWb8UUWRcmk7++UBRB2lxSdlIIs/avgZzsdvQxk8sEQ5qLkqp2Z/LBhMuVtt
10DEhmAoX73zfCKYV5zTj+XXMRWVlZzxZoeslxco60CmwuMRZcMTrolvQJP0oduE
YxlL5G5l4dANoSO9jnb068ANALLK7Zzl3Ku/OM4d6ag4Yzt6EU1RkdVLSXZ66hwu
ACyqsAWlioxfCYDpXmDpjoCvobWap4Ot9Tg+1m4AwItzbRaEdVzPlCRmesvqGnUI
i3xuPwTPaFSBDdvzZHjSyuNYCsSwDokiWRNANwgY0e75eY4fYxUs3LJEpcs8QlgJ
arovBtKa
=W9jc
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `c79fcd31a097ea698807dc5bea3f3e1c7255db211721590c8c645e427a0c180a  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108c79fcd31a097ea698807dc5bea3f3e1c7255db211721590c8c645e427a0c180af010682fe902b021eb53500b514a1f3cf19108fff010866986ee944688adbd017a8619c4b5fa08f1045fca4e74f00892b0207a4ab714990083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0109a00276a02c526c1b0107a59adfddda108f1045fca4e74f008c487746911c507810083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010a524bf2b3dce275b1035f98a06ca430c08f12059fdf20c875269eb7ff5968de8ae78f74704b106a7053f1069ba837f46b35d6608f1045fca4e74f0084303b7e00ddce9950083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010b6e89b1cea7d4e3c24f4a1016b3e900308f1045fca4e74f008ae49f96696b6cff10083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

--- a/test/functional/rpc_signrawtransaction.py
+++ b/test/functional/rpc_signrawtransaction.py
@@ -146,6 +146,19 @@ class SignRawTransactionsTest(BitcoinTestFramework):
         assert_equal(rawTxSigned['errors'][1]['witness'], ["304402203609e17b84f6a7d30c80bfa610b5b4542f32a8a0d5447a12fb1366d7f01cc44a0220573a954c4518331561406f90300e8f3358f51928d43c212a8caed02de67eebee01", "025476c2e83188368da1ff3e292e7acafcdb3566bb0ad253f62fc70f07aeee6357"])
         assert not rawTxSigned['errors'][0]['witness']
 
+    def test_fully_signed_tx(self):
+        self.log.info("Test signing a fully signed transaction does nothing")
+        self.nodes[0].walletpassphrase("password", 9999)
+        self.nodes[0].generate(101)
+        rawtx = self.nodes[0].createrawtransaction([], [{self.nodes[0].getnewaddress(): 10}])
+        fundedtx = self.nodes[0].fundrawtransaction(rawtx)
+        signedtx = self.nodes[0].signrawtransactionwithwallet(fundedtx["hex"])
+        assert_equal(signedtx["complete"], True)
+        signedtx2 = self.nodes[0].signrawtransactionwithwallet(signedtx["hex"])
+        assert_equal(signedtx2["complete"], True)
+        assert_equal(signedtx["hex"], signedtx2["hex"])
+        self.nodes[0].walletlock()
+
     def witness_script_test(self):
         # Now test signing transaction to P2SH-P2WSH addresses without wallet
         # Create a new P2SH-P2WSH 1-of-1 multisig address:
@@ -212,6 +225,7 @@ class SignRawTransactionsTest(BitcoinTestFramework):
         self.script_verification_error_test()
         self.witness_script_test()
         self.test_with_lock_outputs()
+        self.test_fully_signed_tx()
 
 
 if __name__ == '__main__':

